test/js-api/exception/jsTag.tentative.any.js

```diff
@@ -0,0 +1,122 @@
+// META: global=window,dedicatedworker,jsshell,shadowrealm
+// META: script=/wasm/jsapi/assertions.js
+// META: script=/wasm/jsapi/wasm-module-builder.js
+
+test(() => {
+  assert_throws_js(TypeError, () => new WebAssembly.Exception(WebAssembly.JSTag, [{}]))
+}, "Creating a WebAssembly.Exception with JSTag explicitly is not allowed");
+
+promise_test(async () => {
+  const builder = new WasmModuleBuilder();
+  const jsTag = builder.addImportedTag("module", "JSTag", kSig_v_r);
+
+  const throwRefFn = builder.addImport("module", "throw_ref", kSig_r_r);
+  const sig_r_v = builder.addType(kSig_r_v);
+  const kSig_re_v = makeSig([], [kExternRefCode, kExnRefCode]);
+  const sig_re_v = builder.addType(kSig_re_v);
+
+  // Calls throw_ref, catches an exception with 'try_table - catch JSTag', and
+  // returns it
+  builder.addFunction("catch_js_tag_and_return", kSig_r_r)
+    .addBody([
+      kExprBlock, sig_r_v,
+        kExprTryTable, sig_r_v, 1,
+          kCatchNoRef, jsTag, 0,
+          kExprLocalGet, 0,
+          kExprCallFunction, throwRefFn,
+        kExprEnd,
+      kExprEnd,
+    ])
+    .exportFunc();
+
+  // Calls throw_ref, catches an exception with 'try_table - catch_ref JSTag',
+  // and returns it
+  builder.addFunction("catch_ref_js_tag_and_return", kSig_r_r)
+    .addBody([
+      kExprBlock, sig_re_v,
+        kExprTryTable, sig_r_v, 1,
+          kCatchRef, jsTag, 0,
+          kExprLocalGet, 0,
+          kExprCallFunction, throwRefFn,
+        kExprEnd,
+        kExprReturn,
+      kExprEnd,
+      kExprDrop,
+    ])
+    .exportFunc();
+
+  // Calls throw_ref, catches an exception with 'try_table - catch_ref JSTag',
+  // and rethrows it (with throw_ref)
+  builder.addFunction("catch_ref_js_tag_and_throw_ref", kSig_r_r)
+    .addBody([
+      kExprBlock, sig_re_v,
+        kExprTryTable, sig_r_v, 1,
+          kCatchRef, jsTag, 0,
+          kExprLocalGet, 0,
+          kExprCallFunction, throwRefFn,
+        kExprEnd,
+        kExprReturn,
+      kExprEnd,
+      kExprThrowRef,
+    ])
+    .exportFunc();
+
+  function throw_ref(x) {
+    throw x;
+  }
+  const buffer = builder.toBuffer();
+  const {instance} = await WebAssembly.instantiate(buffer, {
+    module: { throw_ref, JSTag: WebAssembly.JSTag }
```

Comment:
should this be `throw_ref: throw_ref,`?

Reply:
That works, but it looks in case the imported name and the external name are the same, just writing it once seems to have the same effect.

Reply:
Yeah, this is the shorthand syntax in JS object initializers.